api/document/js/commands.ts

```diff
@@ -0,0 +1,26 @@
+import { Node } from 'prosemirror-model';
+import { TextSelection } from 'prosemirror-state';
+
+import { factory } from './schema';
+
+export function addParagraph(state, dispatch) {
+  if (!dispatch) {
+    return true;
+  }
+  const pos = state.selection.$head;
+  let depth = pos.depth;
+  while (depth >= 0) {
+    const node = pos.node(depth);
+    if (node.type.spec.group === 'block') {
+      const insertPos = pos.after(depth);
+      let tr = state.tr.insert(insertPos, factory.para(' '));
+      // + 2 to get us inside the inline
+      tr = tr.setSelection(TextSelection.create(tr.doc, insertPos + 2));
+      dispatch(tr.scrollIntoView());
```

Comment:
Do we want to verify that the document still conforms to the schema after the transaction is executed, and roll it back if not?  (Or alternatively, is there a redux-esque-middleware we could add that does this for all transactions, at least during development?)

Reply:
I'll throw in a `.check()` after the dispatch. That'll raise an exception, but should keep chugging along.